openstack/clustering/v1/profiles/results.go

```diff
@@ -49,10 +59,27 @@ type Profile struct {
 	User      string                 `json:"user"`
 }
 
+type ProfilePage struct {
+	pagination.LinkedPageBase
+}
+
+// IsEmpty determines if a ProfilePage contains any results.
+func (page ProfilePage) IsEmpty() (bool, error) {
+	profiles, err := ExtractProfiles(page)
+	return len(profiles) == 0, err
+}
+
 func (r *Profile) UnmarshalJSON(b []byte) error {
+	type tmpSpec Spec
```

Comment:
This technically works, but for consistency, `Spec` should have its own `UnmarshalJSON` method. This is important for two reasons: if a future modification to `Spec` requires further unmarshalling, then the method will already be in place, and it prevents others from copying the style which might lead to bugs down the road.